### ISSUE TITLE
Question: Add legend to columnchart clustered

### ISSUE THREAD

**Issue by iilyak** [johannes-wolf/cetz#240]
Currently the bars within the group do not have labels and legend is not provided. It is hard to tell that the bars are actually represent p99, p95, min, max

<img width="312" src="https://github.com/johannes-wolf/cetz/assets/9804420/974a7e8c-9a2c-426f-9bfb-12548086c1c7">

**Comment by iilyak** [comment]
I was able to add legend, but it would be really cool to have this functionality in CeTZ.

<img width="588" alt="Screenshot 2023-10-04 at 11 35 23 AM" src="https://github.com/johannes-wolf/cetz/assets/9804420/cc296936-275d-48ab-9c9e-4145fb12240f">

```typst
#let legend = (
  "p99": (stroke: black, fill: rgb("#3D5A80")),
  "p95": (stroke: black, fill: rgb("#98C1D9")),
  "min": (stroke: black, fill: rgb("#E0FBFC")),
  "max": (stroke: black, fill: rgb("#EE6C4D")),
)

#let legend-item(point, name, style) = {
  import cetz.draw: *
  content(
    (point.at(0)+2, point.at(1)+0), (point.at(0)+5, point.at(1)+0), 
    par(justify: false)[#name],
    frame: "rect",
    padding: .5em,
    fill: style.fill
  )
}

#let style(index) = legend.values().at(index)

#cetz.canvas({
  cetz.chart.columnchart(
    mode: "clustered",
    y-label: "ms",
    size: (auto, 5),
    label-key: 0,
    value-key: (..range(1, 5  )),
    bar-width: 0.9,
    bar-style: style,
    data3
  )

  let y = 4
  for (key, style) in legend {
    legend-item((4, y), key, style)
    y = y + 0.5
  }
})
```

Specifically what I don't like in the above solution is manual coordinates management.

**Comment by johannes-wolf** [comment]
I want to add a legend function for charts and plots, but I am not yet sure what a good API could look like.

**Comment by reitermarkus** [comment]
> what a good API could look like.

Maybe a `value-labels` option, since this adds labels to each value, and a `legend` option to specify the position of those labels, e.g. `legend: none | "left" | "right" | "bottom"`.

**Comment by johannes-wolf** [comment]
Implemented with #337.

Preview:
![grafik](https://github.com/johannes-wolf/cetz/assets/519002/6630aeda-fc21-4c36-8308-751fff876988)

The chart API will use the plot API and therefore uses the same legend mechanism.
You can freely move the legend around or use some predefined anchors, see `plot.plot` for details.
